From cb50727cfd4186b401b54cbdc8ffc8fd9b4d2c2f Mon Sep 17 00:00:00 2001
From: Jeremy Stashluk <stashlukj@geophysical.com>
Date: Mon, 12 Nov 2018 16:47:13 -0500
Subject: [PATCH] of: overlay: remove sysfs_create_groups call

This crashes the kernel when an runtime overlay is applied.
---
 drivers/of/overlay.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/of/overlay.c b/drivers/of/overlay.c
index 7f8cc5c..a63f624 100644
--- a/drivers/of/overlay.c
+++ b/drivers/of/overlay.c
@@ -828,13 +828,6 @@ static int __of_overlay_create(struct device_node *tree,
 	if (err)
 		goto err_revert_overlay;
 
-	err = sysfs_create_groups(&ov->kobj, ov->attr_groups);
-	if (err != 0) {
-		pr_err("%s: sysfs_create_groups() failed for tree@%s\n",
-				__func__, tree->full_name);
-		goto err_remove_kobj;
-	}
-
 	ov->kobj.kset = ov_kset;
 	err = kobject_add(&ov->kobj, NULL, "%d", id);
 	if (err != 0) {
@@ -852,8 +845,6 @@ static int __of_overlay_create(struct device_node *tree,
 
 	return id;
 
-err_remove_kobj:
-	kobject_put(&ov->kobj);
 err_cancel_overlay:
 	of_changeset_revert(&ov->cset);
 err_revert_overlay:
-- 
2.7.4

